examples/community/pipeline_stable_diffusion_3_controlnet.py

```diff
@@ -0,0 +1,1032 @@
+            negative_prompt_3=negative_prompt_3,
+            do_classifier_free_guidance=self.do_classifier_free_guidance,
+            prompt_embeds=prompt_embeds,
+            negative_prompt_embeds=negative_prompt_embeds,
+            pooled_prompt_embeds=pooled_prompt_embeds,
+            negative_pooled_prompt_embeds=negative_pooled_prompt_embeds,
+            device=device,
+            clip_skip=self.clip_skip,
+            num_images_per_prompt=num_images_per_prompt,
+        )
+
+        if self.do_classifier_free_guidance:
+            prompt_embeds = torch.cat([negative_prompt_embeds, prompt_embeds], dim=0)
+            pooled_prompt_embeds = torch.cat([negative_pooled_prompt_embeds, pooled_prompt_embeds], dim=0)
+
+
+        # 3.1 Prepare control_image
+        controlnet_conditioning = self._prepare_controlnet_conditioning(
+            controlnet_conditioning,
+            width,
+            height,
+            batch_size,
+            num_images_per_prompt,
+            device,
+            dtype,
+            self.do_classifier_free_guidance,
+            False,
+        )
+
+        # 4. Prepare timesteps
+        timesteps, num_inference_steps = retrieve_timesteps(self.scheduler, num_inference_steps, device, timesteps)
+        num_warmup_steps = max(len(timesteps) - num_inference_steps * self.scheduler.order, 0)
+        self._num_timesteps = len(timesteps)
+
+        # 5. Prepare latent variables
+        num_channels_latents = self.transformer.config.in_channels
+        latents = self.prepare_latents(
+            batch_size * num_images_per_prompt,
+            num_channels_latents,
+            height,
+            width,
+            prompt_embeds.dtype,
+            device,
+            generator,
+            latents,
+        )
+
+        # 6. Denoising loop
+        with self.progress_bar(total=num_inference_steps) as progress_bar:
+            for i, t in enumerate(timesteps):
+                if self.interrupt:
+                    continue
+
+                # expand the latents if we are doing classifier free guidance
+                latent_model_input = torch.cat([latents] * 2) if self.do_classifier_free_guidance else latents
+                # broadcast to batch dimension in a way that's compatible with ONNX/Core ML
+                timestep = t.expand(latent_model_input.shape[0])
+
+
+                # controlnet(s) inference
+                if len(controlnet_conditioning) > 0:
+                    control_block_samples_list = []
+
+                    for cc_info in controlnet_conditioning:
+                        # input
+                        control_model_input = latent_model_input
+
+                        # control image
+                        controlnet_conditioning_image = cc_info['control_image']
+
+                        # control encoder_hidden_states
+                        control_encoder_hidden_states = cc_info.get('control_encoder_hidden_states', None)
```

Comment:
let's make them pipeline arguments, e.g. `controlnet_prompt_embeds`
what is `control_encoder_hidden_states` anyway? do we allow user pass a different prompt for controlnet? what's a use case that this is needed?